streams/src/test/java/org/apache/kafka/streams/integration/ProcessingExceptionHandlerIntegrationTest.java

```diff
@@ -78,6 +82,7 @@ public void shouldFailWhenProcessingExceptionOccurs() {
             .map(KeyValue::new)
             .mapValues(value -> value)
             .process(runtimeErrorProcessorSupplierMock())
+            .process(runtimeErrorPunctuateProcessorSupplierMock())
```

Comment:
You should not test punctuate with the `TopologyTestDriver`. Method `punctuate` is called outside of the topology in the poll loop. Using the `TopologyTestDriver`, you basically call `task.maybePunctuateSystemTime();` in `driver.advanceWallClockTime()` which is a helper test infra method. This is not really testing the integration of `ProcessingExceptionHandler` with `punctuate()` in the Streams runtime.
Testing the integration of `ProcessingExceptionHandler` in `process()` with the `TopologyTestDriver` makes a little bit more sense because you can test the case when a downstream processor throws and it should not affect an upstream processor, but that case you do not have with `punctuate()`. However, also for `process()` a real intergration test would be better.
So given the tight schedule, my proposal is to revert the changes regarding `punctuate()` in this file and only have unit tests. Then, you can add a real integration test in a separate PR when we merged all the rest first.